utilities/transactions/pessimistic_transaction.h

```diff
@@ -49,6 +49,9 @@ class PessimisticTransaction : public TransactionBaseImpl {
 
   Status Commit() override;
 
+  // It is basically Commit without going through Prepare phase. The write batch
+  // is also directly provided instead of expecting txn to gradually batch the
+  // transactions writes to an internal write batch.
   virtual Status CommitBatch(WriteBatch* batch) = 0;
```

Comment:
How we are using this API?

Reply:
This is a public API. Users can call it instead of going through Write + Commit steps.

Reply:
I'm wondering why this API ever existing. Fine if we don't know.

Reply:
Git shows that it was there from the very beginning. Probably someone figured it would be an optimization by avoiding the memcpy of the source write batch to the destination write batch.

Reply:
Ah I see. It make sense if it is there prior to 2PC.